src/core/lib/iomgr/timer_manager.cc

```diff
@@ -224,7 +224,11 @@ static bool wait_until(grpc_exec_ctx* exec_ctx, grpc_millis next) {
 static void timer_main_loop(grpc_exec_ctx* exec_ctx) {
   for (;;) {
     grpc_millis next = GRPC_MILLIS_INF_FUTURE;
+    // Check if it is time to update g_start_time
+    grpc_exec_ctx_maybe_update_start_time(exec_ctx);
```

Comment:
Drive by comment: you invalidate at the start of this function, and then again right after. I'd lean towards invalidating right before this function call, and removing the invalidation from maybe_update_start_time